### DISCUSSION TITLE
[v2] How to use `tauri::api::path::home_dir` on tauri v2?

### DISCUSSION THREAD

**Discussion by percy507** [tauri-apps/tauri#7757] [Q&A] [answered]
In tauri v1, I can use `tauri::api::path::home_dir` directly without passing any params. What is the equal code in tauri v2?

```rust
// v1 code
use tauri::api::path::home_dir;
use std::path::PathBuf;

pub fn config_file() -> PathBuf {
  home_dir()
    .expect("failed to resolve home_dir")
    .join(format!(".config.{}", crate::APP_ID))
}
```

**Comment by FabianLars** [comment, marked as answer]
I'm not exactly sure _why_ (maybe it's a bit more tricky on mobile?) but now you need access to an instance of App, AppHandle or Window to get the PathResolver, which in v1 was only necessary for app specific dirs.

As an alternative you can use the `dirs` crate directly which is effectively the same.

**Reply by percy507** [reply]
👍Thanks. https://crates.io/crates/dirs
